examples/chef/sample_app_util/README.md

```diff
@@ -0,0 +1,167 @@
+# Chef Build Conventions
+
+
+## Overview
+
+---
+
+It is convenient to follow some naming and build conventions for Chef tool due to the large
+volume of sample apps that may be created and the ambiguity that may result from arbitrary
+names.
+
+There are three components to the convention proposed here:
+1. The naming convention for the sample matter device and clusters (referred to here as the
+`sample app`).
+2. The naming convention to use for the build files which will be flashed on the devices.
+3. The usage of metadata files that shall accompany build files to provide more detailed
+information about builds.
+
+The convention proposed here should be adopted by the zap files provided in `examples/chef/devices`
+and the builds generated from Chef tool in CI.
+
+## Limitations
+
+---
+
+The largest filename that can be used on MacOS and Linux is 255 characters.
+If a sample app name would exceed this limit by following this convention, then the sample app
+should be given an arbitrary name.
+
+This limitation is called out, but, with the given naming conventions, this should rarely happen.
+
+## Convention
+
+---
+
+### Sample App Naming Convention
+
+Sample apps should be named by concatenating the name of all endpoints in the order of their index.
+Endpoint names are separated by underscores (`_`) and a 10 character hash of the sample app
```

Comment:
what is this 10 character hash? I think it needs some definition - data format and contents.